storage-node/src/services/sync/cleanupService.ts

```diff
@@ -73,8 +72,8 @@ export async function performCleanup(
     getStorageObligationsFromRuntime(qnApi, buckets),
     getDataObjectIDs(),
   ])
-  const assignedObjectsIds = model.dataObjects.map((obj) => obj.id)
-  const removedIds = _.difference(storedObjectsIds, assignedObjectsIds)
```

Comment:
Should I revert this change?, because the problem seems only to be with `differenceWith` implementation, not `difference`

Reply:
Yeah, sure, if there appears to be no problem with the `_. difference` function, then lets have it

Reply:
Yeah it is cleaner with `_.difference`.
Will do.

Reply:
Worth noting, _.difference was still 10x slower than doing filter+Map.has

Also there is use of  `Array.some()` later in the cleanup service.. maybe worth doing some timing tests on that?